.github/workflows/ci-golang-lint.yml

```diff
@@ -19,15 +19,23 @@ jobs:
     name: lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+      - name: Get Go Version
+        run: |
+          #!/bin/bash
+          GOVERSION=$({ [ -f .go-version ] && cat .go-version; })
+          echo "GOVERSION=$GOVERSION" >> $GITHUB_ENV
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GOVERSION }}
       - name: golang-lint
         env:
           # prevent OOM
           GOGC: 10
-        uses: golangci/golangci-lint-action@v2
+        uses: golangci/golangci-lint-action@v3.2.0
         with:
           # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
-          version: v1.36
+          version: v1.48
```

Comment:
also should we just do latest here?

Reply:
latest can work here - currently latest is [v1.50.1](https://github.com/golangci/golangci-lint/releases) so not a big bump

Reply:
it's easier if we pin to latest. We shouldn't be doing `latest` pins on core code though.

The other thing is there may be some `go version` <> `go-lang-ci version` compatibility issues.

Reply:
Also pinned to latest.  👍